src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -319,33 +329,26 @@ cdef class AioServer:
                 break
 
             # Accepts new request from Core
-            rpc_state = await _server_call_request_call(
-                self._server,
-                self._cq,
-                self._loop)
-
-            # Schedules the RPC as a separate coroutine
-            rpc_task = self._loop.create_task(
-                _handle_rpc(
-                    self._generic_handlers,
-                    rpc_state,
-                    self._loop
-                )
-            )
+            rpc_state = await self._request_call()
+
+            # Creates the dedicated RPC coroutine. If we schedule it right now,
```

Comment:
Nice job catching this race condition! Was this causing any tests to fail/flake? If not, can we add one?

Reply:
It is covered by any early cancellation test cases, e.g. [`test_early_cancel_unary_stream`](https://github.com/grpc/grpc/blob/master/src/python/grpcio_tests/tests_aio/unit/call_test.py#L217). After firing the RPC, sends cancel from the client-side immediately. If the RPC coroutine is running and the cancellation coroutine is not, the test will hang.

This race condition is more likely to happen in unit tests because both the client and the server are running in the same event loop.